contracts/PoolManager.sol

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: HORNEY
+
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "./Authorizable.sol";
+import "./interfaces/IRewardPool.sol";
+
+contract PoolManager is Authorizable {
+    using EnumerableSet for EnumerableSet.AddressSet;
+    using SafeERC20 for IERC20;
+
+    EnumerableSet.AddressSet private fullPoolList;
+    EnumerableSet.AddressSet private legacyPoolList;
+    IERC20 public governanceToken;
+
+    event AddPools(address[] pools, bool isLegacy);
+    event AddPool(address pool, bool isLegacy);
+    event RemovePools(address[] pools, bool isLegacy);
+    event GovernanceTokenChange(IERC20 newGovernanceToken);
+
+    constructor(IERC20 _governanceToken) {
+        governanceToken = _governanceToken;
+    }
+
+    /*
+        Write Functions - Permissioned
+    */
+
+    function addPool(address _pool, bool _isLegacy) public onlyAuthorized {
+        if (_isLegacy) {
+            if (!legacyPoolList.contains(_pool)) {
+                legacyPoolList.add(_pool);
+            }
+        } else {
+            if (!fullPoolList.contains(_pool)) {
+                fullPoolList.add(_pool);
```

Comment:
Might need some validation that this is in fact a `GNANA` pool?

Reply:
I don't disagree in a require where `stakeToken == governanceToken`

Reply:
My intent was to have this contract serve 2 purposes:
1. Governance token balance indicator
2. Have a comprehensive list of all pools & jungle farms (for the frontend to eventually consume).

If we limit it to governance token, I think we lose the ability to do 2